src/articles/articles.controller.ts

```diff
@@ -0,0 +1,74 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Put,
+  Delete,
+  UseGuards,
+  Request,
+  Query,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { AuthGuard } from '@nestjs/passport';
+import {
+  ArticlesService,
+  CreateArticleDto,
+  UpdateArticleDto,
+  SingleArticleResponse,
+  ArticlesResponse,
+} from './articles.service';
+import { User } from '@prisma/client';
+import { OptionalAuthGuard } from "../auth/optional-auth.guard";
+import { Request as ExpressRequest } from 'express';
+
+@Controller('articles')
+export class ArticlesController {
+  constructor(private readonly articlesService: ArticlesService) {}
+
+  // POST /api/articles - Create Article
+  @UseGuards(AuthGuard('jwt')) // Yêu cầu đăng nhập
+  @Post()
+  createArticle(
+    @Request() req: ExpressRequest & { user: User },
+    @Body('article') createArticleDto: CreateArticleDto,
+  ) {
+    return this.articlesService.createArticle(req.user, createArticleDto);
```

Comment:
```suggestion
    return this.articlesService.create(req.user, createArticleDto);
```
với service cũng tương tự.
ko cần lặp lại cái đuôi "article"